lib/models/application.coffee

```diff
@@ -678,6 +678,145 @@ getApplicationModel = (deps, opts) ->
+	#
+	# @description Configures the application to run a particular release
+	# and not get updated when the latest release changes.
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	# @param {String} fullReleaseHash - the hash of a successful release (string)
+	# @returns {Promise}
+	#
+	# @example
+	# balena.models.application.pinToRelease('MyApp', 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# balena.models.application.pinToRelease(123, 'f7caf4ff80114deeaefb7ab4447ad9c661c50847').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# balena.models.application.pinToRelease('MyApp', 'f7caf4ff80114deeaefb7ab4447ad9c661c50847', function(error) {
+	# 	if (error) throw error;
+	# 	...
+	# });
+	###
+	exports.pinToRelease = (nameOrId, fullReleaseHash, callback) ->
+		getId(nameOrId)
+		.then (applicationId) ->
+			pine.patch
+				resource: 'application'
+				id: applicationId
+				body:
+					commit: fullReleaseHash
+					should_track_latest_release: false
+		.asCallback(callback)
+
+	###*
+	# @summary Get the hash of the current release for a specific application
+	# @name getTargetReleaseHash
+	# @public
+	# @function
+	# @memberof balena.models.application
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	# @fulfil {String} - The release hash of the current release
+	# @returns {Promise}
+	#
+	# @example
+	# balena.models.application.getTargetReleaseHash('MyApp').then(function(release) {
+	# 	console.log(release);
+	# });
+	#
+	# @example
+	# balena.models.application.getTargetReleaseHash(123).then(function(release) {
+	# 	console.log(release);
+	# });
+	#
+	# @example
+	# balena.models.application.getTargetReleaseHash('MyApp', function(release) {
+	# 	console.log(release);
+	# });
+	###
+	exports.getTargetReleaseHash = (nameOrId, callback) ->
+		exports.get(nameOrId, $select: 'commit')
+		.get('commit')
+		.asCallback(callback)
+
+	###*
+	# @summary Configure a specific application to track the latest available release
+	# @name trackLatestRelease
+	# @public
+	# @function
+	# @memberof balena.models.application
+	#
+	# @description The application's current release will be updated with each new successfully built release.
+	#
+	# @param {String|Number} nameOrId - application name (string) or id (number)
+	# @returns {Promise}
+	#
+	# @example
+	# balena.models.application.trackLatestRelease('MyApp').then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# balena.models.application.trackLatestRelease(123).then(function() {
+	# 	...
+	# });
+	#
+	# @example
+	# balena.models.application.trackLatestRelease('MyApp', function(error) {
+	# 	if (error) throw error;
+	# 	...
+	# });
+	###
+	exports.trackLatestRelease = (nameOrId, callback) ->
+		getId(nameOrId).then (applicationId) ->
+			return pine.patch
+				resource: 'application'
+				id: applicationId
+				body: should_track_latest_release: true
```

Comment:
Found the thread about doing it in an API hook. I guess since v5 is already out, it might be late to expect that to happen soon.
See: https://www.flowdock.com/app/rulemotion/resin-frontend/threads/k5B1N0BEPFHVQJYv-9wdegSs1Ji